graphql/resolvers.go

```diff
@@ -359,6 +360,44 @@ func (r *queryResolver) TaskFiles(ctx context.Context, taskID string) ([]*Groupe
 	}
 	return groupedFilesList, nil
 }
+func (r *queryResolver) TaskLogs(ctx context.Context, taskID string) (*RecentTaskLogs, error) {
+	const LogMessageCount = 100
+	var loggedEvents []event.EventLogEntry
+	// loggedEvents is ordered ts descending
+	loggedEvents, err := event.Find(event.AllLogCollection, event.MostRecentTaskEvents(taskID, LogMessageCount))
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Unable to find EventLogs for task %s: %s", taskID, err.Error()))
+	}
+
+	// remove all scheduled events except the youngest and push to filteredEvents
+	filteredEvents := []event.EventLogEntry{}
+	foundScheduled := false
+	for i := 0; i < len(loggedEvents); i++ {
+		if foundScheduled == false || loggedEvents[i].EventType != event.TaskScheduled {
+			filteredEvents = append(filteredEvents, loggedEvents[i])
+		}
+		if loggedEvents[i].EventType == event.TaskScheduled {
+			foundScheduled = true
+		}
+	}
+
+	// reverse order so ts is ascending
+	for i := len(filteredEvents)/2 - 1; i >= 0; i-- {
```

Comment:
This is clever, but do we actually want chronological order here? I realize the existing UI has it this way, but with 100 events I believe the most interesting events will be last, which will make reverse chronological sorting better

Reply:
this sounds like a good idea, let's bring it up with the UI planning team